src/components/molecules/NavBarScheduleWeather/NavBarScheduleWeather.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect, useState } from "react";
+import classNames from "classnames";
+import { isEqual } from "lodash";
+
+import { getBlackRockWeatherInfo, WeatherInfo } from "api/openweathermap";
+
+import { ContainerClassName } from "types/utility";
+
+import { useInterval } from "hooks/useInterval";
+
+import { BlackRockWeatherBackground } from "components/molecules/NavBarScheduleWeather/WeatherBackground";
+
+import "./NavBarScheduleWeather.scss";
+
+export interface NavBarScheduleWeatherProps extends ContainerClassName {}
+
+const UPDATE_INTERVAL = 30 * 1000;
+
+export const NavBarScheduleWeather: React.FC<NavBarScheduleWeatherProps> = ({
+  containerClassName,
+}) => {
+  const [weather, setWeather] = useState<WeatherInfo>();
+  const fetchWeather = useCallback(async () => {
+    const current = await getBlackRockWeatherInfo();
+    if (!current) setWeather(undefined);
+    setWeather((prev) => (isEqual(prev, current) ? prev : current));
+  }, []);
+
+  useEffect(() => {
+    void fetchWeather();
+  }, [fetchWeather]);
+  useInterval(fetchWeather, UPDATE_INTERVAL);
```

Comment:
> It will stop after the component unmounts, won't it? 

Maybe, I'm asking because I'm not sure how the `useInterval` abstracts and handles the workings of the `setInterval`, not as a matter of rhetoric. Do you have a clear yes/no from a local run or two?

Reply:
yes:)